tools/ddrgen/src/scanners/dwarf/AixDwarfParser.cpp

```diff
@@ -0,0 +1,1034 @@
+			ret = DW_TAG_union_type;	
+			
+			/* Split the name portion into actual name and typeID */
+			str_vect tmp = split (m.prefix(), ':');
+			
+			parsedData->name = tmp[0];
+			parsedData->typeID = tmp[1];
+			parsedData->type = m[0];
+			parsedData->members = m.suffix();
+			parsedData->format = OLD_FORMAT;
+
+			parsedData->type = stripLeading(parsedData->type, '=');
+			parsedData->members = striptrailing(parsedData->members, ';');
+		} else if (regex_search(data, m, classExpression)) {
+			
+			/* Split the name portion into actual name and typeID */
+			str_vect tmp = split (m.prefix(), ':');
+			
+			parsedData->name = tmp[0];
+			parsedData->typeID = tmp[1];
+			parsedData->type = m[0];
+			parsedData->members = m.suffix();
+			parsedData->format = NEW_FORMAT;
+
+			parsedData->type = stripLeading(parsedData->type, '=');
+			parsedData->type = striptrailing(parsedData->type, '(');
+			parsedData->members = striptrailing(parsedData->members, ';');
+			
+			/* Check if the class declaration is for a structure, union or class*/
+			unsigned int index = 1;
+			while ((parsedData->type).length() != index && !isalpha((parsedData->type)[index])) {
+				++index;
+			}
+			if (0 == isCompilerGenerated(parsedData->members)) {
+				if (regex_search (parsedData->type, m, classStructure)) {
+					ret = DW_TAG_structure_type;
+				} else if (regex_search(parsedData->type, m, classUnion)) {
+					ret = DW_TAG_union_type;
+				} else {
+					ret = DW_TAG_class_type;
+				}
+			}else {
+				/* Class declaration is compiler generated, so we ignore it */
+				ret =  DW_TAG_unknown;
+			}
+		} else if (regex_search(data, m, enumExpression)) {
+			ret = DW_TAG_enumeration_type;
+			
+			/* Split the name portion into actual name and typeID */
+			str_vect tmp = split (m.prefix(), ':');
+			
+			parsedData->name = tmp[0];
+			parsedData->typeID = tmp[1];
+			parsedData->type = m[0];
+			parsedData->members = m.suffix();
+
+			parsedData->type = stripLeading(parsedData->type, '=');
+			parsedData->members = stripLeading(parsedData->members, ':');
+			parsedData->members = striptrailing(parsedData->members, ';');
+			
+		}
+	}
+	return ret;
+}
+
+static int
+parseDwarfDie(const string line)
+{
+	
+	int ret = DW_DLV_OK;
+	/* get the type of the data of the line being parsed */
+	Parsed_Declaration_Data *parsedDeclarationData = new Parsed_Declaration_Data();
+	
+	Dwarf_Half tag = parseDeclarationLine(line, parsedDeclarationData);
+	
+	/* Parse the line further based on the type of declaration */
+	if (DW_TAG_unknown != tag) {
+		Dwarf_Die newDie;
+		double typeID = extractTypeID(parsedDeclarationData->typeID);
+		
+		double ID = _fileID + typeID;
+		
+		/* Search for the die in the createdDies before creating a new one */
+		unordered_map<double, Dwarf_Die>::iterator existingEntry = createdDies.find(ID);
+		
+		if (createdDies.end() != existingEntry) {
+			newDie = existingEntry->second;
+		} else {
+			newDie = new Dwarf_Die_s();
```

Comment:
Initialize all fields of Dwarf_Die_s either by setting them individually or using memset() or however you prefer